### ISSUE TITLE
Remove ID prop from auro-button

### ISSUE THREAD

**Issue by geoffrich** [AlaskaAirlines/auro-button#130]
# General Support Request

## Support request
`auro-button` defines an `id` prop that is passed down from the host to the `button` in the shadow root. For example:

```
<auro-button id="test">
  <shadow-root>
    <button id="test"></button>
  </shadow-root>
</auro-button>
```

Why is this prop necessary?  The ID in the shadow root cannot be referenced or queried by anything outside of the shadow root without heavy workarounds. It seems unnecessary to define `id` as a prop at all, since consumers can still add an `id` attribute to `auro-button` without the prop defined.

It is also confusing because ID can be set through the property (e.g. `button.id = "test"`), but this does not reflect to the attribute. If we do want to keep `id` as a property, we should reflect it. However, I can't see what purpose it's currently serving, and think it would be better to remove it. Anyone querying or referencing the ID they gave to `auro-button` is receiving a reference to the host element, not the shadow button, so I don't think this is a breaking change.

This came up as part of https://github.com/AlaskaAirlines/auro-popover/issues/28.

**Comment by blackfalcon** [comment]
You are correct. This is a hold-over from a legacy assumption that we needed this in order to pass events into the shadowDOM. 

I feel that your assumptions are also correct as placing the ID on the outer `auro-button` is the thing that the outer app context would query and removing that from the `<button>` DOM inside [should] not have any effect. But... that is an assumption and we currently have no way of really knowing. 

If we reflect, that is a minor release. If we remove that should be a major release as there is the potential for impact and this alone is not significant for a major release. 

Let's reflect and let's make sure to reference this issue for future opportunity to remove this code.

**Comment by blackfalcon** [comment]
The recommendation is to remove the ID property from the auro-button element scope. Placing an ID on the outer custom element will address any need for query selection of the element. Passing this into the scope of the button element is unnecessary and creates issues that are not necessary to solve other than removing this property. 

This is a BREAKING CHANGE and should be included with the v7.0-rc. 

### Exit criteria

Validate that all current functionality regarding the auro-button is still operable prior to release.

**Comment by blackfalcon** [comment]
See commit for review https://github.com/AlaskaAirlines/auro-button/pull/210/commits/c84f5915f342b9d425d9887e8298ddf30b152765
